consensus/src/block_storage/sync_manager.rs

```diff
@@ -351,25 +422,33 @@ impl BlockStore {
         );
 
         if !order_vote_enabled {
+            // TODO: this is probably still necessary, but need to think harder, it's pretty subtle
             // check if highest_commit_cert comes from a fork
             // if so, we need to fetch it's block as well, to have a proof of commit.
-            let highest_commit_certified_block_id = highest_commit_cert
-                .certified_block(order_vote_enabled)?
-                .id();
+            let highest_commit_certified_block =
+                highest_commit_cert.certified_block(order_vote_enabled)?;
             if !blocks
                 .iter()
-                .any(|block| block.id() == highest_commit_certified_block_id)
+                .any(|block| block.id() == highest_commit_certified_block.id())
             {
                 info!(
                     "Found forked QC {}, fetching it as well",
                     highest_commit_cert
                 );
                 BLOCKS_FETCHED_FROM_NETWORK_WHILE_FAST_FORWARD_SYNC.inc_by(1);
+                let target_block_retrieval_payload = match window_size {
```

Comment:
@zekun000 do you know why the logic here is different for `!order_vote_enabled`?

Reply:
yes, basically when order_vote is disabled, highest_commit_cert has to also be a QuorumCert for a block (which may be a fork that's being discarded), after order vote, we don't view it as QC anymore so don't need to fetch the discarded fork.

think about this chain
B0 <- B1 <- QC (Highest commit cert)
      <- B2 <- B3 <- QC(Highest quorum cert)
      
before order vote we need to fetch B1 in order to insert the highest commit cert to block tree (inserting the QC needs to have the certified block exist first)